### PR TITLE
Update CompilerService to show errors when invalid models are uploaded

### PR DESCRIPTION
**Pull request by mrerrormessage**
See https://github.com/NetLogo/Tortoise/pull/140

### CODE REVIEW

File: app/controllers/CompilerService.scala [hunk 1]
```diff
@@ -58,11 +58,33 @@ object CompilerService extends Controller {
...
+        }
+        failures.tail.foldLeft[\/[NonEmptyList[Exception], NonEmptyList[CompilerException]]](firstFailure) {
+          case (ces: \/-[NonEmptyList[CompilerException]], ce: CompilerException) => ces.map(_.<::(ce))
+          case (ces: \/-[NonEmptyList[CompilerException]], e: Exception)          => \/.left(NonEmptyList(e))
+          case (es: -\/[NonEmptyList[Exception]],          e: CompilerException)  => es
+          case (es: -\/[NonEmptyList[Exception]],          e: Exception)          => es.leftMap(_.<::(e))
+        }.fold(
+            otherExceptions    => otherExceptions.map(_.getMessage).failure,
+            compilerExceptions => compilerExceptions.failure.successNel[String])
+      },
```
Comment (mrerrormessage) [inline app/controllers/CompilerService.scala]: I believe the operation in the preceding 12 lines can be done in fewer lines using casts. Perhaps there's an easier way in scalaz? Basically, we're looking to return any CompilerExceptions while converting any non-compiler-exceptions into strings.

**Comment by TheBizzle** [comment]
Thoughts on [this](https://github.com/NetLogo/Galapagos/commit/0dcd61c459d1f35ba13ff4d3038368715b2c6b0a), @mrerrormessage?

**Comment by mrerrormessage** [comment]
That's much better. I'll update the PR to use that. Thanks for taking a look!

**Comment by mrerrormessage** [comment]
Changes have been applied. Provided Travis-CI passes, this should be ready for further review (if desired) or merge.

**Comment by mrerrormessage** [comment]
I've applied the change and I'm testing. I'll comment again when ready for further review/merge.

**Comment by mrerrormessage** [comment]
OK, as soon as Travis passes, this should be ready for review. Thanks for your help @TheBizzle.

**Pull request merged**